src/System.Private.CoreLib/shared/System/Text/Rune.cs

```diff
@@ -0,0 +1,724 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    /// <summary>
+    /// Represents a Unicode scalar value ([ U+0000..U+D7FF ], inclusive; or [ U+E000..U+10FFFF ], inclusive).
+    /// </summary>
+    /// <remarks>
+    /// This type's constructors and conversion operators validate the input, so consumers can call the APIs
+    /// assuming that the underlying <see cref="Rune"/> instance is well-formed.
+    /// </remarks>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public readonly struct Rune : IComparable<Rune>, IEquatable<Rune>
```

Comment:
I still don't love the name and would have preferred something like UnicodeScalar.  The fact that the XML comments states "Represents a Unicode scalar" just reinforces for me that such a descriptive name would be better for this type, over "Rune" which as far as I can tell is just gibberish, and for which as far as I can tell the best argument for it is that both Go and Swift use it, albeit apparently each with a slightly different actual meaning.  That said, I understand there are two entrenched naming camps here; seems like there are a few very passionate folks in favor of Rune and lots of less passionate folks against but not so much that we're willing to fall on a sword for it.

Reply:
I'm definitely moving more toward "passionate" that it should (have) be(en) UnicodeScalar.  While it's wordier, it's more descriptive, and the things like "GetRunes/EnumerateRunes" just feel... wrong.

> and for which as far as I can tell the best argument for it is that both Go and Swift use it

Swift actually uses unicode scalar. (https://developer.apple.com/documentation/swift/unicode/scalar)

> seems like there are a few very passionate folks in favor of Rune and lots of less passionate folks against but not so much that we're willing to fall on a sword for it. And I understand this is what was agreed to in API review.

I think that @terrajobst preferred Rune over UnicodeScalar due to the shorter name.  And I think I led the majority with "I don't like Rune, but I won't go to the mat over this" and we essentially decided/agreed to not care.

So I'll move to a stronger "I think Rune is a bad name"; but it's possibly too late to matter, and probably still being outweighed by the ones who prefer Rune.

The closest to an FDG rule I can find is "DO favor readability over brevity".  I hereby assert that UnicodeScalar is more readable (in that it's more descriptive).

Reply:
I want to add, Rune is not even showing up in Unicode glossary https://www.unicode.org/glossary/ Also, looking at the dictionary, it has the definitions: 

```
noun
any of the characters of certain ancient alphabets, as of a script used for writing the Germanic languages, especially of Scandinavia and Britain, from c200 to c1200, or a script used for inscriptions in a Turkic language of the 6th to 8th centuries from the area near the Orkhon River in Mongolia.
something written or inscribed in such characters.
an aphorism, poem, or saying with mystical meaning or for use in casting a spell.
 ```

It will be hard to map Unicode Scalar definition to Rune definition. 

I vote for UnicodeScalar